R/tbl-lazy.R

```diff
@@ -201,7 +201,14 @@ distinct_.tbl_lazy <- function(.data, ..., .dots = list(), .keep_all = FALSE) {
 add_op_join <- function(x, y, type, by = NULL, copy = FALSE,
                         suffix = c(".x", ".y"),
                         auto_index = FALSE, ...) {
-  by <- common_by(by, x, y)
+  by_intersect <- intersect(tbl_vars(x), tbl_vars(y))
```

Comment:
This doesn't handle the case where `by` is a named vector that maps the names of the join columns in the left table to the join columns in the right table, as in:
```r
full_join(tbl_left, tbl_right, by = c("xl" = "xr"))
```

Reply:
Thanks for the reference, no change here.